docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-adding-adhoc-tasks-proc.adoc

```diff
@@ -1,2 +1,19 @@
-[id='']
-= TODO
\ No newline at end of file
+[id='case-management-adding-adhoc-tasks-proc-{context}']
+= Adding ad hoc tasks 
+
+Ad hoc tasks are optional user or service tasks that are configured in a case definition. Because they are ad hoc, they must be triggered in some way, usually by a signal event or Java API call.
+
+.Prerequisite 
+
+* A case project has been created in {CENTRAL}.
+* A case definition has been created.
+
+.Procedure 
+. In the case designer, click image:3898.png[right arrows] to open the *Object Library* on the left of the design palette.
+. From the list, select a user or service task that you want to add to the case design as an ad hoc task. 
++
+For example, in the IT Orders sample project, the `Order rejected` user task is an ad hoc task that would only be performed if the request for new hardware is denied by the manager. 
++
+You might also want to add a user task to fill out a warranty claim if goods that are received are faulty. A warranty claim may or may not need to be completed during the course of the case, however it is available to be performed in an ad hoc way if required.
+. Configure the ad hoc task properties in the *Properties* panel on the right.
+. Trigger the task during runtime using a Java API call. For example, the  `Order rejected` ad hoc task is triggered by the following Java API call: `return  KieFunctions.isFalse(approved);`
```

Comment:
this java API call is not correct as it's more for conditions rather than signals. case mgmt comes with dedicated api that exposes number of methods. in this particular case it is CaseService.triggerAdHocFragment